test/cadet/assessments_test.exs

```diff
@@ -0,0 +1,200 @@
+defmodule Cadet.AssessmentsTest do
+  use Cadet.DataCase
+
+  alias Cadet.Assessments
+  alias Cadet.Accounts
+
+  test "all missions" do
+    missions = [
+      insert(:mission),
+      insert(:mission),
+      insert(:mission),
+      insert(:mission),
+      insert(:mission)
+    ]
+
+    result = Assessments.all_missions()
+    assert Enum.all?(result, fn x -> x.id in Enum.map(missions, fn m -> m.id end) end)
+  end
+
+  test "all open missions" do
+    open_mission = insert(:mission, is_published: true, category: :mission)
+    closed_mission = insert(:mission, is_published: false, category: :mission)
+    result = Enum.map(Assessments.all_open_missions(:mission), fn m -> m.id end)
+    assert open_mission.id in result
+    refute closed_mission.id in result
+  end
+
+  test "create mission" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "mission",
+        category: :mission,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    assert mission.title == "mission"
+    assert mission.category == :mission
+  end
+
+  test "create sidequest" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "sidequest",
+        category: :sidequest,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    assert mission.title == "sidequest"
+    assert mission.category == :sidequest
+  end
+
+  test "create contest" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "contest",
+        category: :contest,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    assert mission.title == "contest"
+    assert mission.category == :contest
+  end
+
+  test "create path" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "path",
+        category: :path,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    assert mission.title == "path"
+    assert mission.category == :path
+  end
+
+  test "create programming question" do
+    mission = insert(:mission)
+
+    {:ok, question} =
+      Assessments.create_question(
+        %{
+          title: "question",
+          weight: 5,
+          type: :programming,
+          question: %{},
+          raw_question: "{\"content\": \"asd\", \"solution_template\": \"template\",
```

Comment:
This might look cleaner if you instead do `Poison.encode!(%{content: "asd", solution_template: "template"})`. Similarly for the other `raw_*` fields